mrjob/aws.py

```diff
@@ -158,7 +158,7 @@
 _S3_REGIONLESS_ENDPOINT = 's3.amazonaws.com'
 
 # us-east-1 doesn't have its own endpoint or need bucket location constraints
```

Comment:
Didn't the bucket location constraints go away?  If so, this comment and constant name should be reworked.